# in libclik_mkl.so is created already in configure phase, and it would be deleted
# by Easybuild doing installdir cleanup.
#
# - 1.3.2019, J. Dvoracek (Institute of Physics | Czech Academy of Sciences, www.fzu.cz
#
diff -Nru plc-2.0_orig/waf_tools/any_lapack.py plc-2.0/waf_tools/any_lapack.py
--- plc-2.0_orig/waf_tools/any_lapack.py	2015-06-03 14:54:27.000000000 +0200
+++ plc-2.0/waf_tools/any_lapack.py	2018-12-07 19:04:30.390774000 +0100
@@ -7,6 +7,8 @@
 from waflib import Errors
 import sys
 import waflib
+import shutil	#to be able copy files
+import os #to have mkdir
 
 version = "lapack-3.3.1"
 tool = "lapack-3.3.1"
@@ -87,6 +89,12 @@
       #print llgo
       #print llge
       ctx.end_msg(cmdlist["res"])
+
+      #PATCH: here we should copy the libclik_mkl.so to some safe place
+      if not osp.exists('build/src/'):
+        os.mkdir('build/src/')
+      shutil.copy(cmdlist["res"],'build/src/')
+ 
       ctx.options.lapack_link = "-lclik_mkl "+cmdlist["omp"]
       ctx.options.lapack_lib = ctx.env.LIBDIR+":".join([""]+ctx.env.LIBPATH_fc_runtime)
       ctx.options.lapack_include =  ctx.options.lapack_mkl+"/include"
diff -Nru plc-2.0_orig/waf_tools/execconfig.py plc-2.0/waf_tools/execconfig.py
--- plc-2.0_orig/waf_tools/execconfig.py	2015-05-18 17:22:15.000000000 +0200
+++ plc-2.0/waf_tools/execconfig.py	2018-12-07 16:27:41.917522000 +0100
@@ -70,6 +70,11 @@
   if stlibs:
     alibs += (self.env.STLIB_MARKER or "") +" ".join([srlibpath,stlibs])
 
+
+  # in the original, bindir is not present, as it expects configure and make in same context - that's not true in EB.
+  if not osp.exists(self.env.BINDIR):
+    os.mkdir(self.env.BINDIR)
+
   f=open(osp.join(self.env.BINDIR,self.target),"w")
   print >>f,config_tpl%(" ".join((includepath,defines,cflags)),alibs)
   f.close()  
diff -Nru plc-2.0_orig/wscript plc-2.0/wscript
--- plc-2.0_orig/wscript	2015-05-18 17:30:50.000000000 +0200
+++ plc-2.0/wscript	2018-12-07 18:13:58.079188000 +0100
@@ -470,6 +470,8 @@
     # install the module file. This is a cheap trick... grml
     shutil.copy('build/clik.mod',ctx.env.INCDIR)
     shutil.copy('build/clik_plik.mod',ctx.env.INCDIR)
+    # restore libclik_mkl.so from safe place as we lost it when libdir was cleaned by easybuild
+    shutil.copy('build/src/libclik_mkl.so',ctx.env.LIBDIR)
     # go around a waf bug which set the wrong chmod to fortran exec
     os.chmod("%s/bin/clik_example_f90"%ctx.env["PREFIX"],Utils.O755)
     build_env_files(ctx)
